doc/api/net.md

```diff
@@ -246,168 +352,217 @@ server.on('error', (e) => {
 <!-- YAML
 added: v0.3.4
 -->
 
-Construct a new socket object.
+* `options` {Object}
+  * `fd` {number} Used to specify a numeric file descriptor of an existing
+    socket. Defaults to `null`.
+  * `allowHalfOpen` {boolean} If `true`, allows the socket to be in a "half
+    open" state. Defaults to `false`.
+  * `readable` {boolean} If `fd` is specified, setting `readable` to `true`
+    allows data from the socket to be read. Defaults to `false`.
+  * `writable` {boolean} If `fd` is specified, setting `writabl` to `true`
+    allows data to be written to the socket. Defaults to `false`.
 
-`options` is an object with the following defaults:
-
-```js
-{
-  fd: null,
-  allowHalfOpen: false,
-  readable: false,
-  writable: false
-}
-```
-
-`fd` allows you to specify the existing file descriptor of socket.
-Set `readable` and/or `writable` to `true` to allow reads and/or writes on this
-socket (NOTE: Works only when `fd` is passed).
-About `allowHalfOpen`, refer to `createServer()` and `'end'` event.
-
-`net.Socket` instances are [`EventEmitter`][] with the following events:
+Construct a new `net.Socket` instance.
 
 ### Event: 'close'
 <!-- YAML
 added: v0.1.90
 -->
 
-* `had_error` {Boolean} `true` if the socket had a transmission error.
+The `'close'` event is emitted once the socket is closed and can no longer be
+used to send or receive data.
+
+The listener callback is called with a single boolean `had_error` argument.
+If `true`, then the socket was closed due to a transmission error.
 
-Emitted once the socket is fully closed. The argument `had_error` is a boolean
-which says if the socket was closed due to a transmission error.
+```js
+socket.on('close', (had_error) => {
+  if (had_error)
+    console.log('There was a transmission error!');
+  else
+    console.log('The connection closed normally.');
+});
+```
 
 ### Event: 'connect'
 <!-- YAML
 added: v0.1.90
 -->
 
-Emitted when a socket connection is successfully established.
-See [`connect()`][].
+The `'connect'` event is emitted when a socket connection is successfully
+established. See [`net.connect()`][].
+
+The listener callback is called without passing any arguments.
+
+```js
+socket.on('connect', () => {
+  console.log('The socket is connected!');
+});
+```
 
 ### Event: 'data'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {Buffer}
+The `'data'` event is emitted when data is received.
 
-Emitted when data is received.  The argument `data` will be a `Buffer` or
-`String`.  Encoding of data is set by `socket.setEncoding()`.
-(See the [Readable Stream][] section for more information.)
+The listener callback will be invoked with a single `data` argument that will
+be either a `Buffer` or string (if an encoding is set using
+`socket.setEncoding()`). See the [Readable][] stream documentation for more
+information.
 
-Note that the __data will be lost__ if there is no listener when a `Socket`
+*Note*: __Data will be lost__ if there is no listener when a `net.Socket`
```

Comment:
yeah, but don't they start paused, so they won't emit the data event until someone listens on it, or calls resume()?